### PR TITLE
Add basic nixos module for fish shell

### PR DESCRIPTION
**Pull request by matthiasbeyer**
My approach on #5331 .

Please note that I'm a complete nixos newbie and I don't even know how to test this! I'm also not a fish shell user, actually, but I guess this is a rather trivial patch, so I tried it.

I copied all stuff I thought would be required for this from the zsh module in `nixos/modules/programs/zsh/`.

### CODE REVIEW

File: nixos/modules/programs/fish/fish.nix [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+      enable = mkOption {
+        default = false;
+        description = ''
+          Whenever to configure fish as an interactive shell.
+          '';
+        type = types.bool;
+      };
+
+      shellAliases = mkOption {
```
Comment (lethalman) [inline nixos/modules/programs/fish/fish.nix]: This is unused.

Comment (matthiasbeyer) [inline nixos/modules/programs/fish/fish.nix]: I added a patch for this, though this diff is not outdated by it.

Comment (lethalman) [inline nixos/modules/programs/fish/fish.nix]: What patch?

**Comment by lethalman** [comment]
Tested locally, but you forgot to add fish to the module list. Adding it manually. Thanks.

File: nixos/modules/programs/fish/fish.nix [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        default = config.environment.shellAliases;
+        description = ''
+          Set of aliases for zsh shell. See <option>environment.shellAliases</option>
```
Comment (henrytill) [inline nixos/modules/programs/fish/fish.nix]: This description needs to be corrected.

**Comment by henrytill** [comment]
Apologies for being a little late to the party here, but...

@lethalman How much did you test this?  

`shellInit` `loginShellInit`, and `interactiveShellInit` don't appear to actually set up the proper initialization files, nor does this address the fact that fish is not bash-compatible, so that sourcing `${config.system.build.setEnvironment}` from a fish initialization file won't even work correctly.

**Comment by lethalman** [comment]
@henrytill it can be improved later by fish users, then.

**Comment by henrytill** [comment]
@matthiasbeyer: https://github.com/NixOS/nixpkgs/issues/5331#issuecomment-73924061

**Comment by matthiasbeyer** [comment]
@henrytill Ah, sad. Didn't expect things to be that difficult. Sorry for the fuckup! :cry:

**Pull request merged**